R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,161 @@
+      fg_missed_30_39 = sum(dplyr::between(dist[temp_miss_idx], 30, 39), na.rm = TRUE),
+      fg_missed_40_49 = sum(dplyr::between(dist[temp_miss_idx], 40, 49), na.rm = TRUE),
+      fg_missed_50_59 = sum(dplyr::between(dist[temp_miss_idx], 50, 59), na.rm = TRUE),
+      fg_missed_60_ = sum(dist[temp_miss_idx] >= 60, na.rm = TRUE),
+      fg_made_list = paste(na.omit(dist[temp_made_idx]), collapse = ";"),
+      fg_missed_list = paste(na.omit(dist[temp_miss_idx]), collapse = ";"),
+      fg_blocked_list = paste(na.omit(dist[temp_block_idx]), collapse = ";"),
+      fg_made_distance = sum(dist[temp_made_idx], na.rm = TRUE),
+      fg_missed_distance = sum(dist[temp_miss_idx], na.rm = TRUE),
+      fg_blocked_distance = sum(dist[temp_block_idx], na.rm = TRUE),
+      .groups = "drop"
+    )
+
+
+  # Extra points
+  df_pat <- df_fg_or_pat %>%
+    dplyr::filter(extra_point_attempt == 1) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      pat_made = sum(pat_res == "good", na.rm = TRUE),
+      pat_att = sum(extra_point_attempt, na.rm = TRUE),
+      pat_missed = sum(pat_res == "failed", na.rm = TRUE),
+      pat_blocked = sum(pat_res == "blocked", na.rm = TRUE),
+      pat_pct = round(pat_made / pat_att, 3L),
+      .groups = "drop"
+    )
+
+
+  # The Game Winning kicks distance include up to one value at the weekly level
+  # but can include multiple across the season. This is one way to account for that.
+  # the downside is that the column names change depending on if it is weekly vs
+  # seasonal.
+  if(weekly) {
+    gw_dist_name <- "gwfg_distance"
+  } else {
+    gw_dist_name <- "gwfg_distance_list"
+  }
+
+  # See the above note. I wonder if this shoudl also include field goals that tie
+  # the game but I kept the filter dplyr::between(score_differential, -2, 0) the way
+  # that is was previously. If you do include field goals that send the game into OT,
+  # then you'll probably need to include the gwfg_distance AND gwfg_distance_list columns
+  # in the weekly data
+  # Also, unusual one for me: An in-line if() statement.
+  game_winners <- df_fg_or_pat %>%
+    dplyr::group_by(game_id, team) %>%
+    dplyr::filter(fixed_drive == max(fixed_drive, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(
+      field_goal_attempt == 1,
+      dplyr::between(score_differential, -2, 0)
+    ) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      gwfg_att = n(),
+      !!gw_dist_name := if(weekly) dist else paste(na.omit(dist), collapse = ";"),
+      gwfg_made = sum(fg_res == "made", na.rm = TRUE),
+      gwfg_missed = sum(fg_res == "missed", na.rm = TRUE),
+      gwfg_blocked = sum(fg_res == "blocked", na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  # Prepping data to merge-in player names. Not 100% sure how you wanted this
+  # done so feel free to scrap the way I did this
+  df_player_names <- df_fg_or_pat %>%
```

Comment:
I would `probably` consider using `nflreadr::load_players()` to backfill player names now instead of doing a distinct over the pbp names - those are notably pretty spotty and can change on a whim

Reply:
Thanks for the flag. Updated it and I believe this has been implemented correctly but it is worth double-checking.